partner_identification/models/res_partner.py

```diff
@@ -163,8 +162,10 @@ def _search_identification(self, field_name, category_code,
         Returns:
             list: Domain to search with.
         """
-
-        return [
-            (field_name, operator, value),
+        id_numbers = self.env['res.partner.id_number'].search([
+            ('name', operator, value),
```

Comment:
Nope - `field_name` is actually irrelevant in the context of searching the `res.partner.id_numbers`. This was actually what was causing the infinite loop, because the domain searched back onto itself instead of the `id_numbers.name`